### PR TITLE
feat/RSS-ECOMM-1_13: add Jest to the project

### LINKED ISSUE #17: Issue RSS-ECOMM-1_13: Install and configure Jest or Vitest  😄
Install and configure Jest or Vitest as the testing framework for the project, enabling comprehensive testing of code functionality. 🔧👩‍💻🧪

Acceptance Criteria 📚
Jest or Vitest is installed and configured as the testing framework for the project. ✔️
The project allows for comprehensive testing of code functionality with Jest or Vitest. ✔️🔬

### PR DESCRIPTION
**Pull request by Maksim99745**
#### Install and configure Jest or Vitest
resolve(#17)

RSS-ECOMM-1_13: Install and configure Jest or Vitest 

#### 🤔 This is a ...

- [x] 🌟 New task

#### Description

- **Brief Overview:**
Added Jest testing framework, 2 example tests and plugins.
Plugins:
eslint-plugin-jest,
eslint-plugin-jest-dom

- **Implementation Approach:**

#### Additional Information

- **Screenshots/Links:**
- ** Related Issues:**
I want you to test these changes locally.

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [x] 📝 I have commented my code, particularly in hard-to-understand areas.
- [x] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.
- [x] 👌 Have at least 1 approve.
- [x] 🌳 Branch from `develop`/`release/...` and target to `develop`/`release/...`.

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export default {
+  preset: 'ts-jest',
+  testEnvironment: 'jest-environment-jsdom',
+  transform: {
+    '^.+\\.tsx?$': 'ts-jest',
+    // process `*.tsx` files with `ts-jest`
+  },
+  moduleNameMapper: {
+    '\\.(gif|ttf|eot|svg|png)$': '<rootDir>/test/__ mocks __/fileMock.js',
```
Comment (Gromov-Roman) [inline jest.config.ts]: rename here

File: src/test/mocks/fileMock.js
Comment (Gromov-Roman) [inline src/test/mocks/fileMock.js:1]: please rename to mocks.js

File: jest.config.ts (hunk 1 above)
Comment (Gromov-Roman) [reply jest.config.ts]: test/mocks/mocks.js

File: package.json
Comment (Gromov-Roman) [inline package.json:1]: check libraries with `npm outdated`

File: package.json [hunk 2]
```diff
@@ -18,13 +18,18 @@
     "formatAll": "npm run lint:fix && npm run format",
     "prepare": "husky",
     "tsc": "tsc --noEmit",
-    "pre-push": "npm run lint && npm run prettier && npm run tsc"
+    "pre-push": "npm run lint && npm run prettier && npm run tsc",
```
Comment (JsPowWow) [inline package.json]: Add your new "test" command to the "pre-push" hook as well

File: package.json
Comment (Maksim99745) [reply package.json:1]: image
Do we have to update this?

File: src/test/mocks/fileMock.js
Comment (Maksim99745) [reply src/test/mocks/fileMock.js:1]: Done, thank you

File: jest.config.ts (hunk 1 above)
Comment (Maksim99745) [reply jest.config.ts]: Done, thank you

File: package.json (hunk 2 above)
Comment (Maksim99745) [reply package.json]: Okey, done

File: package.json
Comment (Gromov-Roman) [reply package.json:1]: no, this is ok

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "formatAll": "npm run lint:fix && npm run format",
     "prepare": "husky",
     "tsc": "tsc --noEmit",
-    "pre-push": "npm run lint && npm run prettier && npm run tsc",
+    "pre-push": "npm run lint && npm run prettier && npm run tsc && test",
```
Comment (Gromov-Roman) [inline package.json]: `&& test` works?
I believe it has to be `&& npm run test`

Comment (Maksim99745) [reply package.json]: You absolutely right 😂

**Review by Gromov-Roman — APPROVED** [review]

**Pull request merged**